llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -1450,6 +1482,7 @@ class LoopVectorizationCostModel {
     if (VF.isScalar() || Uniforms.contains(VF))
       return;
     setCostBasedWideningDecision(VF);
+    setVectorizedCallDecision(VF);
```

Comment:
This is a comment for line 1479, since github reviews rather helpfully don't let you comment on the actual line you care about. :face_palm

It looks like the comment above collectUniformsAndScalars is out of date. I think it was already out of date before this change, but now it's even worse! Could you add something to the comment about collecting decisions regarding call vectorisation?

Reply:
I've added to the comment above it. To some extent this doesn't feel like the right place to call it, but I found that I needed to call setVectorizedCallDecision() in every place collectUniformsAndScalars() was, so added it. I do think it should only be called on one path though -- currently there's multiple paths where the functions are called, depending on things like a user-supplied VF. I fixed one case, but it seems like it could be simplified with another refactoring patch.